docs/source/setup-torchrec.rst

```diff
@@ -0,0 +1,136 @@
+
+==================
+Setting up TorchRec
+==================
+
+In this section, we will:
+
+* Understand requirements for using TorchRec
+* Set up an environment to integrate TorchRec
+* Run basic TorchRec code
+
+
+System Requirements
+-------------------
+
+TorchRec is routinely tested on AWS Linux only and should work in similar environments.
+Below demonstrates the compatability matrix that is currently tested:
+
+.. list-table:: TorchRec Compatability
+   :widths: 25 75
+   :header-rows: 0
+
+   * - Python Version
+     - 3.9, 3.10, 3.11, 3.12
+   * - Compute Platform
+     - CPU, CUDA 11.8, CUDA 12.1
+
+Aside from those requirements, TorchRec's core dependencies are PyTorch and FBGEMM.
+If your system is compatible with both libraries generally, then it should be sufficient for TorchRec.
+
+* `PyTorch requirements <https://pytorch.org/get-started/locally/>`_
+* `FBGEMM requirements <https://pytorch.org/FBGEMM/fbgemm-development/BuildInstructions.html>`_
+
+
+Version Compatability
+---------------------
+
+TorchRec and FBGEMM have matching version numbers that are tested together upon release:
+
+* TorchRec 1.0 is compatible with FBGEMM 1.0
+* TorchRec 0.8 is compatible with FBGEMM 0.8
+* TorchRec 0.8 may not be compatible with FBGEMM 0.7
+
+Furthermore, TorchRec and FBGEMM are released only when a new PyTorch release happens.
+Therefore, specific versions of TorchRec and FBGEMM should correspond to a specific PyTorch version:
+
+* TorchRec 1.0 is compatible with PyTorch 2.5
+* TorchRec 0.8 is compatible with PyTorch 2.4
+* TorchRec 0.8 may not be compatible with PyTorch 2.3
+
+Installations
+-------------
+Below we show installations for CUDA 12.1 as an example. For CPU or CUDA 11.8, swap "cu121" for "cpu" or "cu118".
```

Comment:
We could also put these 4 options into tabs like this ([example result)](https://sphinx-design.readthedocs.io/en/latest/tabs.html#synchronised-tabs):
```
.. tab-set::

    .. tab-item:: Stable (pytorch.org)

        .. code-block:: bash
           pip install torch --index-url https://download.pytorch.org/whl/cu121
           pip install fbgemm-gpu --index-url https://download.pytorch.org/whl/cu121
           pip install torchmetrics==1.0.3
           pip install torchrec --index-url https://download.pytorch.org/whl/cu121

    .. tab-item:: Stable (PyPI, only CUDA 12.1)

        .. code-block:: bash
   	   pip install torch
   	   pip install fbgemm-gpu
    	   pip install torchrec
 ```